types/header.d.ts

```diff
@@ -0,0 +1,4 @@
+/**
+ * The header type declaration of `undici`.
+ */
+export type IncomingHttpHeaders = Record<string, string | string[]>;
```

Comment:
any particular reason why `undefined` is not allowed? Now it's a type error to pass `headers` from an incoming http request into `undici`

Reply:
I think it is just a mistake – no particular reason there. Maybe we can open another PR to fix it?

Reply:
cool 👍 I personally can't think of a reason to disallow `undefined` at least